src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/LockFileUtils.cs

```diff
@@ -994,7 +994,7 @@ private static bool GroupHasNonEmptyItems(IEnumerable<LockFileItem> group)
             string assetType,
             MaccatalystFallback maccatalystFallback)
         {
-            var groups = contentItems.FindItemGroups(patternSet).ToList();
+            var groups = contentItems.FindItemGroups(patternSet).AsList();
```

Comment:
This change is unnecessary. There's no functional change here.
LINQ generates enumerates which had to be converted to Lis